### PR TITLE
dealing with API (fetch, post) and add `remove` function

### PR DESCRIPTION
**Pull request by ahmed-al-farouq**
What I did: 
- Refactor my add book and remove book features to persist the changes in the server.
- Refactor the redux code of the application using middleware to make async requests to the API.
- Add book and remove book work in the same way after the refactor.

### CODE REVIEW

File: src/redux/cofigureStore.js [hunk 1]
```diff
@@ -1,14 +1,49 @@
...
+    })
+    .catch((err) => {
+      dispatch(postDataFailed(err));
+    });
+};
+export const deletetBook = (id) => (dispatch) => {
+  axios.delete(`https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/AnQ6KvkQtjFMjmxqi9Wb/books/${id}`, id)
+    .then((res) => {
+      dispatch(removeBook(res.data));
+      dispatch(getBooks());
```
Comment (dasileker) [inline src/redux/cofigureStore.js:41]: - [ ] The API fetch data should be with the usage of async and await like the requirement.

llll

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @ahmed-al-farouq,** 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

- **Please add more details also to your readme file so the repo will look complete.**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/cofigureStore.js (hunk 1 above)
Comment (ahmed-al-farouq) [reply src/redux/cofigureStore.js]: Hi @dasileker,
I did that.
Thanks for the feedback

**Review by BrianSammit — APPROVED** [review]
# STATUS: APROVED :green_circle: :clap: 

Hello @ahmed-al-farouq, 
Your project is complete!  There is nothing else to say other than... Great job 👏 
Congratulations! 🎉 

Now you can merge this branch and move on :1st_place_medal: 
HAPPY CODING AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover: 

well-1

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**